### PR TITLE
Add docs for internal plugin/test log code

### PR DESCRIPTION
**Pull request by hamishwillee**

### CODE REVIEW

File: en/contributing/dev_logging.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Plugin/Test Logging
+
+DroneCore provides an API that developers can use for console logging in plugins and integration tests. 
+
+The API methods display a custom message, prepending a timestamp and the type of log message (e.g. debug) and appending the origin of the message (file and line number).
+
+> **Tip** The API should be considered "internal". It is not exported, or intended, for use in Dronecore apps (and we do not commit to maintaining compatibility in future versions).
```
Comment (hamishwillee) [inline en/contributing/dev_logging.md]: @julianoes Please in particular check that this is a true statement.

File: en/contributing/dev_logging.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#include "../path-to/core/log.h"
+...
+LogDebug() << "command unsupported (" << something.mavlink_command << ").";
+```
+
+## Log Methods
+
+Function | Description | Example
+--- | --- | ---
+`LogDebug()` | Debug messages showing progress and internal state.<br>Debug messages. *These are not displayed in RELEASE builds*. | <code style="color:white; background-color:black;">[04:55:08&#124;Error] This is a debug message (test_file.cpp:26)</code>
```
Comment (hamishwillee) [inline en/contributing/dev_logging.md]: @julianoes For testing I  added the following to some of our example code:
```
#include "../../core/log.h"
LogDebug() << "This is a LogDebug";
LogInfo() << "This is a LogInfo";
LogWarn() << "This is a LogWarn";
LogErr() << "This is a LogErr";
```
For whatever reason, the DebugInfo output did not show up - I was using a debug build of DroneCore and I "tried" to do a debug build of my code, so this surprised me.

File: en/contributing/dev_logging.md (hunk 1 above)
Comment (julianoes) [reply en/contributing/dev_logging.md]: Yes, I agree with this.

File: en/contributing/dev_logging.md (hunk 2 above)
Comment (julianoes) [reply en/contributing/dev_logging.md]: I've tried this with default cmake settings:

```
﻿mkdir build && cd build
cmake ..
make && ./takeoff_and_land
...
[10:21:38|Info ] This is a LogInfo (takeoff_and_land.cpp:23)
[10:21:38|Warn ] This is a LogWarn (takeoff_and_land.cpp:24)
[10:21:38|Error] This is a LogErr (takeoff_and_land.cpp:25)
```

Then specifying release:

```
cmake -DCMAKE_BUILD_TYPE=Release ..
make && ./takeoff_and_land

[10:24:25|Info ] This is a LogInfo (takeoff_and_land.cpp:23)
[10:24:25|Warn ] This is a LogWarn (takeoff_and_land.cpp:24)
[10:24:25|Error] This is a LogErr (takeoff_and_land.cpp:25)
```

Then specifiying debug:
```
cmake -DCMAKE_BUILD_TYPE=Debug ..
make && ./takeoff_and_land

[10:24:54|Debug] This is a LogDebug (takeoff_and_land.cpp:22)
[10:24:54|Info ] This is a LogInfo (takeoff_and_land.cpp:23)
[10:24:54|Warn ] This is a LogWarn (takeoff_and_land.cpp:24)
[10:24:54|Error] This is a LogErr (takeoff_and_land.cpp:25)
```

Comment (hamishwillee) [reply en/contributing/dev_logging.md]: @julianoes Ah, so the interesting point here is that apps build release by default while DroneCore builds debug by default ...
I'll update the app guide with something on this, but merge this particular PR.

Thanks for the review.

**Pull request merged**